stripe/test/test_resources.py

```diff
@@ -208,6 +208,60 @@ def test_retrieve(self):
         self.assertResponse(res)
 
 
+class ListObjectTests(StripeApiTestCase):
```

Comment:
I didn't bother to go down that road until I received feedback. I wanted to ensure the core devs were ok with the __iter__ recalling the endpoint. It could be reasoned that iterating the list object should only go through the single page.